### ISSUE TITLE
Can't sort on TIMESTAMP column on a cross shard query

### ISSUE THREAD

**Issue by tirsen** [youtube/vitess#3143]
If I try to sort a cross shard query on a TIMESTAMP column I get the following error:
```
types are not comparable: TIMESTAMP vs TIMESTAMP
```

**Comment by sougou** [comment]
TL;DR: I think it's a worthy trade-off to just perform string comparisons of date/time columns. This should work correctly (mostly) even for ordering.

Gory details:
For reference, here is the description on how date/time values are handled in MySQL: https://dev.mysql.com/doc/refman/5.7/en/datetime.html. The more significant thing to note is the bug reported by @sjmudd https://bugs.mysql.com/bug.php?id=83852.

The above documentation basically states that it's not possible to be 100% accurate about date/time comparisons. However, applications are aware of these shortcomings and are willing to tolerate the rare corner cases where things don't work as expected.

I looked at parsing a such strings into a time.Time before the comparison. However, the only improvement it offers over string comparison is for string representations that are intentionally non-canonical. In a way, stating that we compare date/time values as strings makes it less confusing.

**Comment by tirsen** [comment]
That's an excellent compromise. I think for all these things we should just do whatever MySQL does and MySQL prioritizes convenience and 99% correctness over 100% correctness any day of the week. ;-)
